### PR TITLE
Automate updation

### PR DESCRIPTION
**Pull request by thealphadollar**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---

The PR aims to automate the process of updating expected result files for each of the sample present in the regression tests. It precisely brings the following new features and changes:

- Add flask-script.
- Setup manage.py
	- Manage.py is required so that we can handle operations that required the app context.
	- We can now call scripts from inside flask environment.
	- Makes the app more flexible to future development.
- Improve RegressionTestOutput initialisation method as it used to save in sample_id instead of regression_id
- Improve sample_db.py by adding new column for scaffolding RegressionTestOutput as well.

NOTE: Local testing has been done.

### CODE REVIEW

**Comment by thealphadollar** [comment]
@canihavesomecoffee Please review.

File: manage.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    name = 'update'
+    capture_all_args = True
+
+    def run(self, remaining):
+        """Driver function for update subcommand."""
+        if len(remaining) == 0:
+            print('path to ccextractor is missing')
+            return 1
+        else:
```
Comment (canihavesomecoffee) [inline manage.py]: This else is unneeded as you return in the if. Removes a level of indent and makes it a bit easier on the remaining structure too

File: manage.py [hunk 2]
```diff
@@ -28,15 +28,15 @@ def run(self, remaining):
         if len(remaining) == 0:
             print('path to ccextractor is missing')
             return 1
+
+        path_to_ccex = remaining[0]
+        print('path to ccextractor: ' + str(path_to_ccex))
+
+        if update_expected_results(path_to_ccex):
```
Comment (canihavesomecoffee) [inline manage.py]: It's a bit of nitpicking, but this if-else can (just as the previous one) be simplified further :)

File: mod_regression/update_regression.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+        try:
+            proc = subprocess.run(proc_args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            with open(log_file, 'w+') as logf:
+                logf.write(proc.stdout.decode('utf-8'))
+            if proc.returncode != 0:
+                print(f'ERROR: ccextractor encountered error for {input_file}, please see {log_file}')
+                print(f'ccextractor was run as {" ".join(proc_args)}')
+                return False
+            else:
```
Comment (canihavesomecoffee) [inline mod_regression/update_regression.py]: Else can be removed; return True instead :)

**Comment by thealphadollar** [comment]
@canihavesomecoffee Done.

**Pull request merged**